src/khepri_machine.erl

```diff
@@ -1207,27 +1215,29 @@ apply(
   Meta,
   #register_projection{pattern = PathPattern, projection = Projection},
   State) ->
-    Tree = get_tree(State),
+    ProjectionName = Projection#khepri_projection.name,
     ProjectionTree = get_projections(State),
-    Reply = khepri_projection:init(Projection),
-    State1 = case Reply of
-                 ok ->
-                     restore_projection(Projection, Tree, PathPattern),
-                     ProjectionTree1 = khepri_pattern_tree:update(
-                                         ProjectionTree,
-                                         PathPattern,
-                                         fun (?NO_PAYLOAD) ->
-                                                 [Projection];
-                                             (Projections) ->
-                                                 [Projection | Projections]
-                                         end),
-                     erase(compiled_projection_tree),
-                     set_projections(State, ProjectionTree1);
-                 _  ->
-                     State
-             end,
-    Ret = {State1, Reply},
-    bump_applied_command_count(Ret, Meta);
+    case has_projection(ProjectionTree, ProjectionName) of
+        true ->
+            Ret = {State, {error, exists}},
+            bump_applied_command_count(Ret, Meta);
+        false ->
+            ProjectionTree1 = khepri_pattern_tree:update(
+                                ProjectionTree,
+                                PathPattern,
+                                fun (?NO_PAYLOAD) ->
+                                        [Projection];
+                                    (Projections) ->
+                                        [Projection | Projections]
+                                end),
+            erase(compiled_projection_tree),
```

Comment:
I didn't pay attention to this line before. Could you please add a comment to explain what is the `compiled_projection_tree` and it is set?

Reply:
I added a function that wraps the `erlang:erase/1` call and some docs docs and put it next to `get_compiled_projection_tree/1` for extra clarity